substratecommon.go

```diff
@@ -0,0 +1,817 @@
+package substratecommon
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"log"
+	"net/rpc"
+	"os"
+	"os/exec"
+
+	"github.com/hashicorp/go-plugin"
+	"github.com/sirupsen/logrus"
+
+	hclog "github.com/hashicorp/go-hclog"
+)
+
+// ConcreteRequestOptions is a variant of RequestOptions that is
+// "flattened" to pure data.
+type ConcreteRequestOptions struct {
```

Comment:
Do these request options need to be exposed publicly ?

Reply:
I guess yes for the same reason (it's passed through RPC).